runtime-light/tl/tl-core.h

```diff
@@ -9,16 +9,16 @@
 #include <iterator>
 #include <optional>
 #include <string_view>
+#include <utility>
 
-#include "common/mixin/not_copyable.h"
 #include "runtime-common/core/utils/kphp-assert-core.h"
 #include "runtime-light/allocator/allocator.h"
 #include "runtime-light/core/std/containers.h"
 #include "runtime-light/utils/concepts.h"
 
 namespace tl {
 
-class TLBuffer final : private vk::not_copyable {
+class TLBuffer final {
```

Comment:
I'm almost sure it's not necessary to extend `TLBuffer`'s API for now. You may want to rethink the way you use it in your code in case you're facing troubles with current API

Reply:
What is problem of allowing moving TLBuffer?

Reply:
We thus make it easier for `TLBuffer` to outlive a data it views.

Reply:
Sorry, I was thinking about that with another `TLBuffer`'s design in my mind. I want to get rid of `TLBuffer::store_bytes` or at least limit its usage. I think that it's not necessary to store bytes in order to fetch some TL type from it. It should be enough to just init `TLBuffer` with a view on byte sequence

Reply:
This class is called TL_Buffer_, so I assume that it is owner of all its memory. If I got you right, you wanna builder-like behaviour. I think it should be separate class

Reply:
BTW, such a TLBuilder in general case cannot outlive async operation because views inside it can become changed

Reply:
Ok. There are 2 cases for now:

1. add move-semantics to `TLBuffer`
2. just pass it as reference (which is safe in your case)

Reply:
I think mone semantics is useful in anycase.

In my case I prefer move semantics because really that function semantically  does not guarantee anything about tl buffer. That is why initially there were double ref.

Reply:
Why?
What's wrong with awaitables' move assignment?
And with TLBuffer's move assignment operator?

Reply:
I just don't want to add additional functionality unless it's necessary

Reply:
Fixed